Warn about preloaded modules loaded before trace

matthewloring opened the pull request:
Fixes #196

matthewloring commented:
@ofrobots PTAL.

ofrobots commented on README.md:
```diff
@@ -24,10 +24,16 @@ This module provides Cloud Trace support for Node.js applications. [Google Cloud
+
+        node --require @google/cloud-trace app.js
+
+or at the *top of the main script of your application*:
```
I am afraid this is a bit too confusing. Specially the 'see below' part. If the `--require` is our preferred mechanism we should give complete instructions here. If that is too verbose, then `--require` should *not* be our preferred mechanism.

matthewloring replied:
I've taken another pass at this.

ofrobots commented on README.md:
```diff
@@ -24,10 +24,16 @@ This module provides Cloud Trace support for Node.js applications. [Google Cloud
         require('@google/cloud-trace').start({projectId: 'your-project-id'});
 
+It's important that Cloud Trace is the first thing executed so that it can accurately gather data. If you require multiple modules as part of your start up command, cloud trace must be the first since they are loaded in order.
+
```
I think it would be better to state this *before* the mechanism to load is described.

matthewloring replied:
How does this read after the rewrite?

matthewloring commented:
@ofrobots PTAL.

pmuellr commented:
As someone familiar with using this type of tooling, wording looks good to me.  I assume the path to the config file can be relative, and if so would be relative to the current working directory (expected behavior). If true, doesn't require any further clarification, but if not, then clarify further.

matthewloring commented:
@pmuellr Thank you for the suggestion! I agree that config files should be loaded relative to the current working directory. Tracked in #198.

ofrobots commented:
I would prefer if we added a new section in README.md on the different ways a user can pass configuration to the agent and left the default getting started instructions as they are. That keeps things simpler for new users.

I am not too worried anymore about the problem of the user using having some other code loaded through `--require` before the agent is started because you just added a check to detect that.

matthewloring commented:
@ofrobots I've reverted the quick start section of the README and added a new section on configuration. I've omitted the specialized environment variables (log level, etc) so that we don't get locked into those names. If you'd like these to be documented as well I can add them.

matthewloring commented:
@ofrobots Updated. Ready for a look.

ofrobots commented on README.md:
```diff
@@ -30,6 +30,14 @@ This module provides Cloud Trace support for Node.js applications. [Google Cloud
+
+         require('@google/cloud-trace').start({projectId: 'your-project-id', samplingRate: 500});
+
+Alternatively, you can provide configuration through a config file. This can be useful if you want to load our module using --require on the command line instead of editing your main script. You can start by copying the default config file and modifying it to suit your needs. The `GCLOUD_DIAGNOSTICS_CONFIG` environment variable should point to your configuration file.
```
backticks around `--require`.

matthewloring replied:
Done.

ofrobots commented on lib/hooks/index.js:
```diff
@@ -109,6 +106,20 @@ function activate(agent) {
     }
   }
+  if (process._preload_modules && process._preload_modules.length > 0) {
+    var fst = process._preload_modules[0];
```
`fst`: unnecessary abbreviation.

matthewloring replied:
[Habit](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html).

ofrobots replied:
You mean hbt?

ofrobots commented:
LGTM w/ comments.
